### PR TITLE
Use retry mechanism in async version of Connection objects

### PR DESCRIPTION
**Pull request by szumka**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `$ tox` pass with this change (including linting)?
- [x] Do the CI tests pass with this change (enable it first in your forked repo and wait for the github action build to finish)?
- [x] Is the new or changed code fully tested?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Is there an example added to the examples folder (if applicable)?
- [x] Was the change added to CHANGES file?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### Description of change

The retry mechanism was not used in an async version of Connection(and SentinelManagedConnection). I've added the use of 'call_with_retry' - the same as in the sync version. 
See related issue: https://github.com/redis/redis-py/issues/2211

### CODE REVIEW

File: redis/asyncio/connection.py [hunk 1]
```diff
@@ -816,7 +820,7 @@ async def on_connect(self) -> None:
             if str_if_bytes(await self.read_response()) != "OK":
                 raise ConnectionError("Invalid Database")
 
-    async def disconnect(self) -> None:
+    async def disconnect(self, *args) -> None:
```
Comment (dvora-h) [inline redis/asyncio/connection.py:823]: @szumka Why did you add the args here? they have no use in the function

Comment (szumka) [reply redis/asyncio/connection.py:823]: Thanks for the review. I will remove those args.

**Review by dvora-h — APPROVED** [review]

**Comment by szumka** [comment]
@dvora-h is it possible to rerun CI checks? "CI / Validate building and installing the package (who)" hung for 6hrs and was canceled. I don't think the issue is related to this PR.
